src/main/java/duke/taskmanager/TaskManager.java

```diff
@@ -0,0 +1,222 @@
+
+    /**
+     * Returns the date format to be used to allocate task by the task manager.
+     *
+     * @return date format used by the task manager.
+     */
+    public String getDateFormat() {
+        return DATE_FORMAT;
+    }
+
+    /**
+     * Processes strings that are formatted to be readable for task manager.
+     * Used to process strings loaded from a save file creates a new task from the string.
+     *
+     * @param formattedString string in a readable format detailing a task to be added
+     * @return a new task with details processed from the string
+     * @throws Exception when the formatted string cannot be read
+     */
+    private Task processFormattedString(String formattedString) throws Exception {
+        String[] arguments = formattedString.split("<>");
+        String taskType = arguments[0];
+        boolean isCompleted = (arguments[1].equals("1"));
+        String taskName = arguments[2];
+
+        switch (taskType) {
+        case "T":
+            return new ToDoTask(taskName, isCompleted);
+        case "D":
+            return new DeadlineTask(taskName, arguments[3], isCompleted, DATE_FORMAT);
+        case "E":
+            return new EventTask(taskName, arguments[3], isCompleted, DATE_FORMAT);
+        default:
+            return new EmptyTask();
+        }
+    }
+
+    /**
+     * List the task currently in its task list. Responds with an appropriate message
+     * depending on the number of items in the list.
+     *
+     * @return response message detailing the task present in the task list
+     */
+    public String list() {
+        if (taskList.size() == 0) {
+            return "\tYou have no tasks in your list.\n";
+        } else {
+            StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.append("\tI have your list of tasks displayed below:\n");
+            for (int i = 0; i < this.taskList.size(); i++) {
+                stringBuilder.append("\t").append(i + 1).append(") ").append(taskList.get(i)).append("\n");
+            }
+            return stringBuilder.toString();
+        }
+    }
+
+    /**
+     * Adds a new task to the current task list. Responds with a message indicating the
+     * addition of the new task.
+     *
+     * @param task task to be added to the task list
+     * @return response message indicating the addition of the new task
+     */
+    public String addTask(Task task) {
+        this.taskList.add(task);
+        return ("\t> Added: " + task.getTaskName() + "\n");
+    }
+
+    /**
+     * Marks a task in the task list to indicate completion. The task to be marked is provided by the user.
+     * Responds with an appropriate message indicating the marking of the task depending on the
+     * prior completion status of the task
+     *
+     * @param itemNumber index of the task to be marked as completed
+     * @return response message indicating the marking of the task
+     */
+    public String mark(int itemNumber) {
```

Comment:
Consider making this method name more descriptive to show that it's not a getter. The same can be said for the `unmark` method.